docs/relationship_synthesis.md

```diff
@@ -0,0 +1,278 @@
+
+4. Wait for our team to review the pull request and iterate on the feedback.
+
+5. Once approved, merged to the main branch and a new release is issued, it is available for you to use.
+
+6. Enjoy your new relationship synthesis rule being applied automatically to your data! :tada:
+
+# How to configure a new Candidate Relationship
+
+Relationship synthesis involves a two-phase system to determine and create relationships based on the provided rules.
+
+In the first phase, the system verifies if a data point should be considered for creating a relationship by performing duck typing.
+This means checking if the data point has all the required attributes and values specified by the rules.
+
+In the second phase, the data point is used to create a relationship according to the defined rule,
+which specifies the relationship type (guid to guid, candidate, or proposal)
+and how the source and target GUIDs should be created.
+
+Example of a valid relationship definition:
+
+```yaml
+relationships:
+  - name: extServiceCallsExtPixieDns
+    version: 1
+    origins:
+      - OpenTelemetry
+    conditions:
+      - attribute: eventType
+        anyOf: [ "MetricRaw" ]
+      - attribute: entity.type
+        anyOf: [ "PIXIE_DNS" ]
+    relationship:
+      expires: P75M
+      relationshipType: CALLS
+      source:
+        buildGuid:
+          account:
+            attribute: accountId
+          domain:
+            value: EXT
+          type:
+            value: SERVICE
+          identifier:
+            fragments:
+              - attribute: service.name
+            hashAlgorithm: FARM_HASH
+      target:
+        extractGuid:
+          attribute: entity.guid
+```
+
+The `name` property provides information during debugging and can be used for feature flags or logging purposes.
+
+The `origins` contains a list of keys indicating where the telemetry information originates from in a user-friendly way.
+
+The `conditions` section specifies the attributes and matching criteria that determine if the rule applies to a given data point.
+
+The `relationship` section defines various aspects of the relationship, such as its expiration timeframe, the relationship type, 
+and how to extract the source and target GUIDs.
+
+All these sections are required for a rule to be considered valid.
+
+## Version
+
+The `version` field indicates the rule version. This allows for introducing new breaking changes to the formats 
+without requiring immediate updates to all engines. 
+The engine discards any unsupported versions.
+
+## Origin
+
+The `origin` field represents a closed list of values that indicate the source of telemetry. When defining a rule, a list of these values must be provided.
+
+Example:
+
+```yaml
+- origin:
+    - APM Metrics
+    - Open Telemetry
+    - Prometheus
+```
+
+The full list of supported origins (as of July 2023) is the following:
+
+| **Origin source**      | **Description**                                                   |
```

Comment:
@manuelrodriguez-nr please make sure we are including validations on the accepted values from this list.

This comment is a continuation of the previous thread: https://github.com/newrelic/entity-definitions/pull/1138/files#r1265186215